contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -6391,6 +6579,690 @@ bbf_ExecDropStmt(DropStmt *stmt)
+	}
+
+	/* Find the partition scheme used to create partitioned table. */
+	partition_scheme_used_for_table = get_partition_scheme_for_partitioned_table(dbid, logical_schema_name, relname);
+	pfree(logical_schema_name);
+
+	/* partition_scheme_used_for_table will be null for non-partitioned table */
+	if (!partition_scheme_used_for_table ||
+			strcmp(partition_scheme_name, partition_scheme_used_for_table) != 0)
+	{
+		pfree(partition_scheme_name);
+		if (partition_scheme_used_for_table)
+			pfree(partition_scheme_used_for_table);
+		return false;
+	}
+
+	/*
+	 * Column specified with partition scheme should be part of partitioning columns.
+	 * Otherwise, the index is unaligned.
+	 */
+	rel = RelationIdGetRelation(relid);
+	key = RelationGetPartitionKey(rel);
+	partnatts = get_partition_natts(key);
+
+	for (i = 0; i < partnatts; i++)
+	{
+		if (attnum == get_partition_col_attnum(key, i))
+			break;
+	}
+
+	RelationClose(rel);
+
+
+	pfree(partition_scheme_name);
+	if (partition_scheme_used_for_table)
+		pfree(partition_scheme_used_for_table);
+
+	if (i == partnatts) /* not part of partitioning columns */
+		return false;
+	return true;
+}
+
+/*
+ * rename_table_update_bbf_partitions_name
+ *	For a rename operation on a babelfish partitioned table, it renames all partition
+ *	of the table to the new name using hash based on the new name, so that
+ *	name of partitions of any partitioned table doesn't conflict.
+ */
+static void
+rename_table_update_bbf_partitions_name(RenameStmt *stmt)
+{
+	char		*physical_schema_name = stmt->relation->schemaname;
+	char		*logical_schema_name;
+	char		*table_name = stmt->relation->relname;
+	List		*partition_names = NIL;
+	int16		dbid;
+	char		*unique_hash;
+	PlannedStmt	*wrapper;
+	RenameStmt	*rename_partition_stmt = NULL;
+	List		*parsetree;
+	Relation	relation;
+	SysScanDesc	scan;
+	ScanKeyData	key;
+	Oid		parentrelid;
+	Oid		inhrelid;
+	HeapTuple	tuple;
+	const char	*old_dialect;
+	StringInfoData	query;
+	char		*partition_name;
+	char		*new_partition_name;
+
+	logical_schema_name = (char *) get_logical_schema_name(physical_schema_name, true);
+
+	if (!logical_schema_name) /* not a tsql schema */
+		return;
+	
+	dbid = get_dbid_from_physical_schema_name(physical_schema_name, false);
+	
+	if (!is_bbf_partitioned_table(dbid, logical_schema_name, table_name))
+	{
+		pfree(logical_schema_name);
+		return;
+	}
+
+	parentrelid = get_relname_relid(stmt->newname, get_namespace_oid(physical_schema_name, false));
+
+	relation = table_open(InheritsRelationId, AccessShareLock);
+
+	ScanKeyInit(&key,
+			Anum_pg_inherits_inhparent,
+			BTEqualStrategyNumber, F_OIDEQ,
+			ObjectIdGetDatum(parentrelid));
+
+	scan = systable_beginscan(relation, InheritsParentIndexId, true,
+					NULL, 1, &key);
+
+	while ((tuple = systable_getnext(scan)) != NULL)
+	{
+		inhrelid = ((Form_pg_inherits) GETSTRUCT(tuple))->inhrelid;
+		partition_names = lappend(partition_names, get_rel_name(inhrelid));
```

Comment:
get_rel_name returns palloc'd copy so we should clean the list using list_free_deep api

Reply:
We have already freed each of names individually in next for loop.